apps/api/src/app/integrations/e2e/update-integration.e2e.ts

```diff
@@ -36,48 +37,126 @@ describe('Update Integration - /integrations/:integrationId (PUT)', function ()
     expect(integration.credentials.secretKey).to.equal(payload.credentials.secretKey);
   });
 
-  it('should deactivate other providers on the same channel', async function () {
-    const firstProviderPayload = {
-      providerId: 'sendgrid',
+  it('should not allow to update the integration with same identifier', async function () {
+    const identifier2 = 'identifier2';
+    const integrationOne = await integrationRepository.create({
+      name: 'Test1',
+      identifier: 'identifier1',
+      providerId: EmailProviderIdEnum.SendGrid,
+      channel: ChannelTypeEnum.EMAIL,
+      active: false,
+      _organizationId: session.organization._id,
+      _environmentId: session.environment._id,
+    });
+    await integrationRepository.create({
+      name: 'Test2',
+      identifier: identifier2,
+      providerId: EmailProviderIdEnum.SendGrid,
+      channel: ChannelTypeEnum.EMAIL,
+      active: false,
+      _organizationId: session.organization._id,
+      _environmentId: session.environment._id,
+    });
+    const payload = {
+      identifier: identifier2,
+      check: false,
+    };
+
+    const { body } = await session.testAgent.put(`/v1/integrations/${integrationOne._id}`).send(payload);
+
+    expect(body.statusCode).to.equal(400);
+    expect(body.message).to.equal('Integration with identifier already exists');
+  });
+
+  it('should not allow to activate the integration without the credentials', async function () {
+    const integrationOne = await integrationRepository.create({
+      name: 'Test1',
+      identifier: 'identifier1',
+      providerId: EmailProviderIdEnum.SendGrid,
       channel: ChannelTypeEnum.EMAIL,
-      credentials: { apiKey: '123', secretKey: 'abc' },
+      active: false,
+      _organizationId: session.organization._id,
+      _environmentId: session.environment._id,
+    });
+
+    const payload = {
       active: true,
       check: false,
     };
-    const secondProviderPayload = {
-      providerId: 'mailgun',
+
+    const { body } = await session.testAgent.put(`/v1/integrations/${integrationOne._id}`).send(payload);
+
+    expect(body.statusCode).to.equal(400);
+    expect(body.message).to.equal('The credentials are required to activate the integration');
+  });
+
+  it('should allow updating the integration with just identifier', async function () {
+    const integrationOne = await integrationRepository.create({
+      name: 'Test',
+      identifier: 'identifier',
+      providerId: EmailProviderIdEnum.SendGrid,
       channel: ChannelTypeEnum.EMAIL,
-      credentials: { apiKey: '123', secretKey: 'abc' },
       active: false,
+      _organizationId: session.organization._id,
+      _environmentId: session.environment._id,
+    });
+
+    const payload = {
+      identifier: 'identifier2',
```

Comment:
Based on the test name shouldn't we remove the property `check`? Or is it needed?

Reply:
We might need to change then the default logic. As it is, seems if we don't pass anything the validation of credentials gets executed. Might be a breaking change? 😓

Reply:
as far as I remember we want to do the validation by default